src/explorer/CosmosDBTreeItem.ts

```diff
@@ -120,4 +117,89 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
     public hasMoreChildrenImpl(): boolean {
         return false;
     }
+
+    private detectMongoConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
+        const result: IDetectedConnection[] = [];
+        for (const key of Object.keys(appSettings)) {
+            const value = appSettings[key];
+            if (/^mongodb[^:]*:\/\//i.test(value)) {
+                result.push({
+                    keys: [key],
+                    connectionString: appSettings[key]
+                });
+            }
+        }
+        return result;
+    }
+
+    private detectDocDBConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
+        const result: IDetectedConnection[] = [];
+        const regexp = new RegExp(`(.+)${this._endpointSuffix}`, 'i');
+        for (const key of Object.keys(appSettings)) {
+            const match = key && key.match(regexp);
+            if (match) {
+                const prefix = match[1];
+                const endpointKey = prefix + this._endpointSuffix;
+                const keyKey = prefix + this._keySuffix;
+                const documentEndpoint: string | undefined = appSettings[endpointKey];
+                const masterKey: string | undefined = appSettings[keyKey];
+
+                if (documentEndpoint && masterKey) {
+                    const keys: string[] = [endpointKey, keyKey];
+                    let connectionString = `AccountEndpoint=${documentEndpoint};AccountKey=${masterKey};`;
+
+                    const databaseKey = prefix + this._databaseSuffix;
+                    if (Object.keys(appSettings).find((k) => k === databaseKey)) {
+                        keys.push(databaseKey);
+                        connectionString += `Database=${appSettings[databaseKey]}`;
```

Comment:
Is this required in the connectionString?

Reply:
Database name is not required. You can see in Matt's sample [here](https://github.com/fiveisprime/node-todo/blob/cosmos/app/models/todo.js#L5), he actually uses `process.env.AZURE_COSMOS_DATABASE_ID || 'todos'`